kernel/src/common/hardwareinfo.c

```diff
@@ -28,17 +30,20 @@ BoardType detect_boardtype() {
 void init_hardwareinfo() {
     BoardType boardType = detect_boardtype();
     size_t peripheral_base_address;
+    size_t peripheral_region_size;
     switch (boardType) {
         case RaspBerryPiTwo:
             peripheral_base_address = BCM2836_PERIPHERALS_PHYSICAL_BASE;
+            peripheral_region_size = 21 * Mebibyte;
             break;
         default:
             FATAL("Peripheral address for board type not implemented");
     }
     hardware_info = (HardwareInfo){
         .cpuType = ARM1176,
         .boardType = boardType,
-	.peripheral_base_address = peripheral_base_address,
+        .peripheral_base_address = peripheral_base_address,
+	.peripheral_region_size = peripheral_region_size,
```

Comment:
Though this is *very* raspi specific, that's what we're developing for so it's fine

Reply:
Indeed, in certain boards there might be various disjoint regions of MMIO. However, since this information is only used by this file, extending this functionality later shouldn't be an issue.